src/app/fdctl/run/tiles/metrics.c

```diff
@@ -0,0 +1,392 @@
+  ERR_load_crypto_strings();
+}
+
+ulong
+append_datapoints( char *       append_buffer,
+                   ulong        append_buffer_capacity,
+                   char * const tile_name,
+                   ulong        idx,
+                   Datapoint *  datapoints,
+                   ulong        datapoints_sz ) {
+  metrics_tag_t tags[] = {
+      { .key   = "host_id",
+         .value = "" },     /* strncpy below from fd_log_host() */
+      { .key   = "tile",
+          .value = "" },    /* snprintf below */
+      { .key   = "index",
+          .value = "" },    /* snprintf below */
+  };
+  strncpy(  tags[ 0 ].value, fd_log_host(), sizeof( tags[ 0 ].value ) - 1 );
+  strncpy(  tags[ 1 ].value, tile_name,     sizeof( tags[ 1 ].value ) - 1 );
+  snprintf( tags[ 2 ].value, sizeof( tags[ 2 ].value ), "%ld", idx );
+
+  long ts = fd_log_wallclock();
+  for( ulong i=0; i<datapoints_sz; i++ ) {
+    char data[ 4096 ];
+    int ret = fd_metrics_format( data, sizeof( data ), tile_name, tags, sizeof( tags )/sizeof( tags[0] ), &datapoints[i], 1, ts++ );
+    if( FD_UNLIKELY( ret<0 || ret>=(int)sizeof( data ) ) ) {
+      FD_LOG_WARNING(( "snprintf error %d", ret ));
+      return 0;
+    }
+    ulong line_sz = strlen( data ) + 1;
+    if( FD_UNLIKELY( append_buffer_capacity <= line_sz ) ) {
+      FD_LOG_WARNING(( "append_buffer_capacity %lu <= %lu", append_buffer_capacity, strlen( data ) + 1 ));
+      return 0;
+    }
+    strncat( append_buffer, data, append_buffer_capacity - 1 );
+    append_buffer_capacity -= line_sz;
+  }
+  return append_buffer_capacity;
+}
+
+void
+ssl_send_data( const char * _server,
+               ushort port,
+               const char * user_pass,
+               const char * db,
+               char * post_buffer,
+               const ulong post_buffer_sz,
+               char * const data ) {
+  int ssl_sockfd = socket( AF_INET, SOCK_STREAM, 0 );
+  if( FD_UNLIKELY( ssl_sockfd < 0 ) ) {
+    FD_LOG_WARNING(( "socket error %d (%d) %s", ssl_sockfd, errno, strerror( errno ) ));
+    return;
+  }
+
+  /* Resolve the server address */
+  struct hostent * server = gethostbyname( _server );
+  if( FD_UNLIKELY( !server ) ) {
+    FD_LOG_WARNING(( "gethostbyname error %d h_errno=%d", ssl_sockfd, h_errno ));
+    goto cleanup0;
+  }
+  struct sockaddr_in server_addr;
+  memset( &server_addr, 0, sizeof(server_addr) );
+  server_addr.sin_family = AF_INET;
+  server_addr.sin_port = htons( port );
+  memcpy( &server_addr.sin_addr.s_addr, server->h_addr, ( size_t )server->h_length );
+
+  /* Connect to the server */
+  struct sockaddr sockaddr;
+  memcpy( &sockaddr, &server_addr, sizeof( struct sockaddr ) );
+  int ret = connect( ssl_sockfd, &sockaddr, sizeof( server_addr ) );
+  if( FD_UNLIKELY( ret < 0 ) ) {
+    FD_LOG_WARNING(( "connect error %d (%d) %s", ret, errno, strerror( errno ) ));
+    goto cleanup0;
+  }
+
+  /* Create an SSL context */
+  SSL_CTX * ctx = SSL_CTX_new( TLS_client_method() );
```

Comment:
These kinds of things can be initialized once at tile boot

Reply:
Updated.